src/context/WalletProvider/Coinbase/components/Connect.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import type { RouteComponentProps } from 'react-router-dom'
+import type { ActionTypes } from 'context/WalletProvider/actions'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { setLocalWalletTypeAndDeviceId } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { logger } from 'lib/logger'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import type { LocationState } from '../../NativeWallet/types'
+import { CoinbaseConfig } from '../config'
+const moduleLogger = logger.child({ namespace: ['Connect'] })
+
+export interface CoinbaseSetupProps
+  extends RouteComponentProps<
+    {},
+    any, // history
+    LocationState
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+export const CoinbaseConnect = ({ history }: CoinbaseSetupProps) => {
```

Comment:
we'll want to get history from a hook like this

https://github.com/shapeshift/web/blob/develop/src/components/Modals/NativeOnboarding/components/EasyToUse.tsx#L20

Reply:
as discussed this is a straight copypasta from metamask/keplr implementations of this era so i'm fine with this